wallet-new/src/Cardano/Wallet/Kernel/DB/Read.hs

```diff
@@ -55,11 +68,43 @@ accountAvailableUtxo :: DB -> HdAccountId -> Utxo
 accountAvailableUtxo snapshot accountId
     = walletQuery' snapshot (Spec.queryAccountAvailableUtxo accountId)
 
+-- | Returns the available balance for the input 'HdAccountId'.
+accountAvailableBalance :: DB -> HdAccountId -> Coin
+accountAvailableBalance snapshot accountId
+    = walletQuery' snapshot (Spec.queryAccountAvailableBalance accountId)
+
 -- | Returns the total balance for this 'HdAccountId'.
 accountTotalBalance :: DB -> HdAccountId -> Coin
 accountTotalBalance snapshot accountId
     = walletQuery' snapshot (Spec.queryAccountTotalBalance accountId)
 
+-- | Returns the total balance for this 'HdAccountId'.
+accountAddresses :: DB -> HdAccountId -> IxSet HdAddress
+accountAddresses snapshot accountId
+    = walletQuery' snapshot (readAddressesByAccountId accountId)
+
 -- | Returns the total balance for this 'HdAccountId'.
 hdWallets :: DB -> HdWallets
 hdWallets snapshot = snapshot ^. dbHdWallets
+
+-- | Lookup the given 'Address' in the 'HdAccount' current checkpoint.
+-- This getter returns a 'Maybe' because in case of freshly-generated
+-- addresses (which has to be yet incorporated into a block) this lookup might
+-- fail, but upstream consumers might want to deal with this gracefully (for
+-- example in case a new prestine account is populated with many addresses and
+-- the same account is requested by the web API).
```

Comment:
I'm not sure that this `Maybe` is needed. `AddressMeta` is explicitly allowed to be empty. I think you can return`mempty`, because it ought to have a `Monoid` instance (and if it doens't, then the existing code should be refactored slightly so that it does use the `Monoid` interface).

Reply:
Roger.